### PR TITLE
Chakraborty Model

### PR DESCRIPTION
**Pull request by sepandhaghighi**
#### What does this implement/fix? Explain your changes.
- Chakraborty dynamic model added

#### Any other comments?
Document Link : https://nbviewer.jupyter.org/github/ECSIM/opem/blob/chakraborty1/Documents/Chakraborty.ipynb
EXE Link : https://www.dropbox.com/s/ed0q76ysbdnlp20/OPEM_PR_Chakraborty.exe?dl=0

### CODE REVIEW

**Comment by sepandhaghighi** [comment]
@sadrasabouri @sarminh @kasraaskari
Please review this PR ;-)

Document Link : https://nbviewer.jupyter.org/github/ECSIM/opem/blob/chakraborty1/Documents/Chakraborty.ipynb
EXE Link : https://www.dropbox.com/s/ed0q76ysbdnlp20/OPEM_PR_Chakraborty.exe?dl=0

**Review by kasraaskari — APPROVED** [review]

File: opem/Dynamic/Chakraborty.py [hunk 1]
```diff
@@ -0,0 +1,458 @@
...
+
+def Nernst_Gain_Calc(T, I):
+    """
+    Calculate Nernst gain.
+
+    :param T: cell operation temperature [K]
+    :type T : float
+    :param I: cell load current [A]
+    :type I : float
+    :return: Nernst gain [V} as float
```
Comment (sadrasabouri) [inline opem/Dynamic/Chakraborty.py:109]: shouldn't be `[v]` or `{v}`?

File: opem/Dynamic/Chakraborty.py [hunk 2]
```diff
@@ -0,0 +1,458 @@
...
+def Efficiency_Calc(Vcell, u, N):
+    """
+    Calculate PEM cell efficiency.
+
+    :param Vcell: cell voltage [V]
+    :type Vcell:float
+    :param u: fuel utilization  ratio
+    :type u: float
+    :param N0: number of fuel cells in the stack
+    :type N0 : int
```
Comment (sadrasabouri) [inline opem/Dynamic/Chakraborty.py:164]: Despite `N0` here, we have`N` in arguments.

File: opem/Dynamic/Chakraborty.py [hunk 3]
```diff
@@ -0,0 +1,458 @@
...
+                "K": B1,
+                "VE": Estimated_V}
+    except Exception:
+        if TestMode:
+            return {
+                "Status": False,
+                "Message": "[Error] " +
+                Simulation_Title +
+                " Simulation Failed!(Check Your Inputs)"}
+        print("[Error] " +Simulation_Title +" Simulation Failed!(Check Your Inputs)")
```
Comment (sadrasabouri) [inline opem/Dynamic/Chakraborty.py:458]: Spaces are not well-formed in this line.

**Review by sadrasabouri — CHANGES_REQUESTED** [review]
Minor edits may be needed.
Also the number of newlines between each function is not identical in `Chakraborty.py`, it may be needed to be fixed.

**Review by sarminh — APPROVED** [review]
All changes are approved.

File: opem/Dynamic/Chakraborty.py (hunk 1 above)
Comment (sepandhaghighi) [reply opem/Dynamic/Chakraborty.py]: fixed in https://github.com/ECSIM/opem/pull/149/commits/85442649ec3014cd4951649eab558bc36b492b15

File: opem/Dynamic/Chakraborty.py (hunk 2 above)
Comment (sepandhaghighi) [reply opem/Dynamic/Chakraborty.py]: fixed in https://github.com/ECSIM/opem/pull/149/commits/fa88049c82fe84f33ba1b0cc5b36b48bc32dbc7a

File: opem/Dynamic/Chakraborty.py (hunk 3 above)
Comment (sepandhaghighi) [reply opem/Dynamic/Chakraborty.py]: fixed in https://github.com/ECSIM/opem/pull/149/commits/f11754fe594f52e04df770562c29229e311a5ed2

**Review by sadrasabouri — APPROVED** [review]
LGTM

**Pull request merged**